assets/src/scripts/components/hydrograph/date-controls.spec.js

```diff
@@ -0,0 +1,235 @@
+        },
+        requests: {
+            'current:P7D': {
+                timeSeriesCollections: ['coll1']
+            },
+            'compare:P7D': {
+                timeSeriesCollections: ['coll2', 'col4']
+            }
+        },
+        variables: {
+            '45807197': {
+                variableCode: {
+                    value: '00060'
+                },
+                oid: '45807197',
+                variableName: 'Test title for 00060',
+                variableDescription: 'Test description for 00060',
+                unit: {
+                    unitCode: 'unitCode'
+                }
+            },
+            '45807190': {
+                variableCode: {
+                    value: '00010'
+                },
+                oid: '45807190',
+                unit: {
+                    unitCode: 'unitCode'
+                }
+            }
+        },
+        methods: {
+            'method1': {
+                methodDescription: 'method description'
+            }
+        }
+    },
+    statisticsData : {
+        median: {
+            '00060': {
+                '1234': [
+                    {
+                        month_nu: '2',
+                        day_nu: '20',
+                        p50_va: '40',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }, {
+                        month_nu: '2',
+                        day_nu: '21',
+                        p50_va: '41',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }, {
+                        month_nu: '2',
+                        day_nu: '22',
+                        p50_va: '42',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }
+                ]
+            }
+        }
+    },
+    timeSeriesState: {
+        currentVariableID: '45807197',
+        currentDateRange: 'P7D',
+        requestedTimeRange: null,
+        showSeries: {
+            current: true,
+            compare: true,
+            median: true
+        },
+        loadingTSKeys: []
+    },
+    ui: {
+        width: 400
+    }
+};
+
+
+describe('date-controls', () => {
+
+    let div;
+    let store;
+
+    beforeEach(() => {
+        div = select('body').append('div');
+        store = configureStore(TEST_STATE);
+        div.call(provide(store))
+            .call(drawDateRangeControls, '12345678');
+    });
+
+    afterEach(() => {
+        div.remove();
+    });
+
+    it('Expects the date range controls to be created', () => {
+        let dateRangeContainer = select('#ts-daterange-select-container');
+        let customDateDiv = select('div#ts-customdaterange-select-container');
+
+        expect(dateRangeContainer.size()).toBe(1);
```

Comment:
It's a little weird that you would use .size(). The select function only ever returns one element so it either finds it or it doesn't. I believe if it doesn't find the element, the variable would be null. Maybe that's a better test to use.

Reply:
Mary and I did a little sleuthing and determined this is an ok way to check for the existence of the date control container given that we are using d3.select.